nservicebus/outbox/index.md

```diff
@@ -79,8 +79,60 @@ In order to gradually convert an entire system from the DTC to the Outbox:
 WARNING: When verifying outbox functionality, it can sometimes be helpful to temporarily [stop the MSDTC Service](https://technet.microsoft.com/en-us/library/cc770732.aspx). This ensures that the Outbox is working as expected, and no other resources are enlisting in distributed transactions.
 
 
+## Message identity and idempotent processing
+
+If handlers are idempotent then outbox might not be required at all. When implementating idempotent processing keep in mind that any generated message *must* apply the same message identifier if invoked more-then-once to have deduplication working on the infrastructure level or to solve this at the logical level by validating message data. [Read more on how to set a message identifier.](/nservicebus/messaging/message-identity.md)
```

Comment:
If handlers are idempotent than outbox is definitely not required. Also, if a handler is idempotent then message IDs don't really matter e.g. is a handler is adding items to a set then it does not care about message IDs.

Deterministic assignment of message IDs is important if users are doing custom ID-based deduplication.

Reply:
@SzymonPobiega I mention this as often customer are unaware about the message sending/publishing side of things. Very often the data operation is safe to multiple invocations but not the messages sending part. The fact that in additional invocations the handler generates the same logical messages but with a different message identifier is I think not what you want and you always want that to be atleast 'static' as to what the outbox does or be deterministic. You are saying basically only do idempotent processing based on the logical message and ignore the message identifier if I'm not mistaken.

Reply:
@ramonsmits you only need stable/deterministic message IDs if you are doing ID-based deduplication (e.g. Outbox or some infra). Otherwise you don't care (e.g. when you deduplicate based on hash of a body)